src/main/java/frc/robot/subsystems/endEffector/EndEffectorSubsystem.java

```diff
@@ -38,6 +40,21 @@ public EndEffectorSubsystem() {
    * @param voltage the voltage to set to
    * @return set the motor to the voltage
    */
+  public boolean coralLoaded() {
+    // double distance = m_colorSensor.getProximity();
+    double distance = m_coral_detector.getRange();
+    if (m_coral_detector.getStatus() != TimeOfFlight.Status.Valid) {
+      DogLog.log("EndEffector/Validity", m_coral_detector.getStatus());
+      DogLog.log("EndEffector/Distance", distance);
+      return false;
+    }
+    if (distance < 30 && distance > 20) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
```

Comment:
```suggestion
  public boolean coralLoaded() {
    return endEffector.IO.coralLoaded();
  }
```

I hope this makes more sense essentially